packages/core/src/testInteg/perf/prepareRepoData.test.ts

```diff
@@ -40,35 +50,44 @@ function performanceTestWrapper(numFiles: number, fileSize: number) {
             const telemetry = new TelemetryHelper()
             return {
                 setup: async () => {
-                    return await createTestWorkspace(numFiles, {
+                    const fsSpy = sinon.spy(fs)
+                    const workspace = await createTestWorkspace(numFiles, {
                         fileNamePrefix: 'file',
                         fileContent: getRandomString(fileSize),
                         fileNameSuffix: '.md',
                     })
+                    return { workspace, fsSpy, numFiles, fileSize }
                 },
-                execute: async (workspace: WorkspaceFolder) => {
-                    return await prepareRepoData([workspace.uri.fsPath], [workspace], telemetry, {
+                execute: async (setup: setupResult) => {
+                    return await prepareRepoData([setup.workspace.uri.fsPath], [setup.workspace], telemetry, {
                         record: () => {},
                     } as unknown as Span<AmazonqCreateUpload>)
                 },
-                verify: async (_w: WorkspaceFolder, result: resultType) => {
-                    verifyResult(result, telemetry, numFiles * fileSize)
+                verify: async (setup: setupResult, result: resultType) => {
+                    verifyResult(setup, result, telemetry, numFiles * fileSize)
                 },
             }
         }
     )
 }
 
-function verifyResult(result: resultType, telemetry: TelemetryHelper, expectedSize: number): void {
+function verifyResult(setup: setupResult, result: resultType, telemetry: TelemetryHelper, expectedSize: number): void {
     assert.ok(result)
     assert.strictEqual(Buffer.isBuffer(result.zipFileBuffer), true)
     assert.strictEqual(telemetry.repositorySize, expectedSize)
     assert.strictEqual(result.zipFileChecksum.length, 44)
+
+    assert.ok(getFsCallsUpperBound(setup.fsSpy) <= setup.numFiles * 4, 'total system calls should be under 4 per file')
```

Comment:
I see a lot of different numbers that we are comparing the upper bound to. Out of curiosity how did you determine the number to use?